### PR TITLE
chore: update deps

### PR DESCRIPTION
**Pull request by Stebalien**
Hopefully this will let us use fewer duplicate deps.

Explicitly doesn't update:

- bellperson
- neptune
- bitvec 1.0 because I couldn't get it working and don't really care (only used in testing).

Importantly:

- Upgrades to clap 3
- Upgrades to bitvec 1
- Upgrades to blake2b/blake2s 1.0
- Upgrades byte-unit (requires rust 2021 edition support).
- Replaces cpuid-bool with cpufeatures (as recommended).
- Upgrades aes and fixes some fallout.
- Upgrades hwloc (in case that matters)
- Upgrades num-bigint (in case that matters)

### CODE REVIEW

File: fil-proofs-tooling/src/bin/gpu-cpu-test/main.rs [hunk 1]
```diff
@@ -34,11 +34,24 @@ const POST_CONFIG: PoStConfig = PoStConfig {
...
-    pub enum Mode {
-        Threads,
-        Processes,
+#[derive(Debug, Clone)]
+pub enum Mode {
+    Threads,
+    Processes,
+}
+
+impl FromStr for Mode {
```
Comment (Stebalien) [inline fil-proofs-tooling/src/bin/gpu-cpu-test/main.rs:43]: Technically, one can derive `ArgEnum`. But that doesn't actually work with `type_of_t`, so I'm not sure what one is supposed to do here. But this version _should_ actually work.

File: fr32/src/padding.rs [hunk 2]
```diff
@@ -793,23 +793,23 @@ mod tests {
             let num_bits = rng.gen_range(1..data.len() * 8 - pos);
             let new_offset = rng.gen_range(0..8);
 
-            let mut bv = BitVec::<LittleEndian, u8>::new();
+            let mut bv = BitVec::<u8, Lsb0>::new();
```
Comment (Stebalien) [inline fr32/src/padding.rs:796]: I'm pretty sure my bitfield changes are correct, but double check.

File: storage-proofs-core/src/crypto/aes.rs [hunk 3]
```diff
@@ -9,17 +10,19 @@ const IV: [u8; 16] = [0u8; 16];
...
     ensure!(key.len() == 32, "invalid key length");
 
-    let mode = Cbc::<Aes256, ZeroPadding>::new_var(key, &IV).context("invalid key")?;
+    let mode = cbc::Decryptor::<aes::Aes256>::new_from_slices(key, &IV).context("invalid key")?;
 
-    let res = mode.decrypt_vec(ciphertext).context("failed to decrypt")?;
+    let res = mode
+        .decrypt_padded_vec_mut::<ZeroPadding>(ciphertext)
+        .context("failed to decrypt")?;
     Ok(res)
```
Comment (Stebalien) [inline storage-proofs-core/src/crypto/aes.rs:26]: Check this carefully.

**Comment by Stebalien** [comment]
This turned into a bit of a "deal with all the deprecated things"-fest.

File: fr32/src/padding.rs (hunk 2 above)
Comment (Stebalien) [reply fr32/src/padding.rs]: EH, I dropped this.

**Comment by Stebalien** [comment]
> Hopefully this will let us use fewer duplicate deps.

Ha, ha ha. Who was I kidding.

**Comment by Stebalien** [comment]
The test failures are timeouts. I'm not sure what the right approach there is.

**Comment by vmx** [comment]
> The test failures are timeouts. I'm not sure what the right approach there is.

I'm working on that. :( that our current cache is broken. Sounds like I need to provide a quick fix to unblock folks.

**Comment by vmx** [comment]
@Stebalien https://github.com/filecoin-project/rust-fil-proofs/pull/1582 that PR should make CI work again. Sadly it takes a long time to run (I'm also working on that problem).

File: fil-proofs-tooling/Cargo.toml [hunk 4]
```diff
@@ -30,30 +30,31 @@ bellperson = "0.18.0"
...
 anyhow = "1.0.23"
 ff = "0.11.0"
 rand_xorshift = "0.3.0"
 bytefmt = "0.1.7"
 rayon = "1.3.0"
-flexi_logger = "0.16.1"
+flexi_logger = "0.22.3"
 typenum = "1.11.2"
 generic-array = "0.14.4"
 # `byte-unit = 4.0.13` is the final version that uses Rust 2018 edition.
```
Comment (vmx) [inline fil-proofs-tooling/Cargo.toml:50]: This comment should be removed.

**Review by vmx — COMMENTED** [review]
LGTM, but I'd like that @DrPeterVanNostrand has another at the hasher changs.

**Comment by cryptonemo** [comment]
> LGTM, but I'd like that @DrPeterVanNostrand has another at the hasher changs.

Ditto, thanks!

**Comment by cryptonemo** [comment]
Closing in favor of #1584

**Pull request closed without merging**